### PR TITLE
moved changes to new branch

### PR DESCRIPTION
**Pull request by trumko**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "movie",
+    "mark"
+  ],
+  "author": "Mykhailo",
+  "license": "ISC",
+  "dependencies": {
+    "babel-jest": "^24.5.0",
+    "react-redux": "^6.0.1",
+    "react-router": "^4.3.1",
+    "react-test-renderer": "^16.8.4",
```
Comment (blyzniuk) [inline package.json]: `babel-jest` and `react-test-renderer` are dev dependencies

File: package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "@babel/preset-react": "^7.0.0",
+    "babel-loader": "^8.0.5",
+    "css-loader": "^2.1.1",
+    "enzyme": "^3.9.0",
+    "enzyme-adapter-react-16": "^1.11.2",
+    "enzyme-to-json": "^3.3.5",
+    "html-webpack-plugin": "3.2.0",
+    "jest": "^24.5.0",
+    "react": "^16.8.4",
+    "react-dom": "^16.8.4",
```
Comment (blyzniuk) [inline package.json]: `react` and `react-dev` are not dev dependencies

File: src/сomponents/App/App.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from "react";
+import { render } from "react-dom";
+import FirstType from "../Experimental/FirstType/FirstType";
+import { SecondType } from "../Experimental/SecondType";
+import ThirdType from "../Experimental/ThirdType";
+import FourthType from "../Experimental/FourthType";
+
+export default class App extends Component {
+  constructor() {
```
Comment (blyzniuk) [inline src/сomponents/App/App.js]: Why do you need a constructor here?

File: src/сomponents/App/App.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from "react";
+import { render } from "react-dom";
```
Comment (blyzniuk) [inline src/сomponents/App/App.js]: Why do you need a render here?
It should be used once, when you render app to the DOM (`src/index.js`)

File: src/сomponents/Experimental/FirstType/FirstType.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Adapter from 'enzyme-adapter-react-16';
+import { shallow, configure } from 'enzyme';
+import { shallowToJson } from 'enzyme-to-json';
+import FirstType from './FirstType';
+
+configure({adapter: new Adapter()});
```
Comment (blyzniuk) [inline src/сomponents/Experimental/FirstType/FirstType.test.js]: adapter should be configured only once. Good practice to do it in separate jest-set-up file

File: src/сomponents/Experimental/FirstType/FirstType.test.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+describe('FirstType', () => {
+  it('should render correctly', () => {
+    const output = shallow(
+      <FirstType>First title</FirstType>
+    );
+    expect(shallowToJson(output)).toMatchSnapshot();
```
Comment (blyzniuk) [inline src/сomponents/Experimental/FirstType/FirstType.test.js]: You could add `enzyme-to-json/serializer` to jest config. Then you will no need to write `shallowToJson` everywhere

File: src/сomponents/Experimental/FirstType/FirstType.test.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  });
+  it('should render correctly', () => {
```
Comment (blyzniuk) [inline src/сomponents/Experimental/FirstType/FirstType.test.js]: The assertion should have a unique name. To specify what are you testing

**Review by blyzniuk — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (trumko) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (trumko) [reply package.json]: fixed

File: src/сomponents/App/App.js (hunk 3 above)
Comment (trumko) [reply src/сomponents/App/App.js]: I don't need it, I think I saw some tutorial, where adding constructor was considered as "best practice". I guess I was wrong

File: src/сomponents/App/App.js (hunk 4 above)
Comment (trumko) [reply src/сomponents/App/App.js]: Removed, thank you! I forgot to clear it after refactoring and moving the render logic to`src/index.js`

File: src/сomponents/Experimental/FirstType/FirstType.test.js (hunk 5 above)
Comment (trumko) [reply src/сomponents/Experimental/FirstType/FirstType.test.js]: fixed

File: src/сomponents/Experimental/FirstType/FirstType.test.js (hunk 6 above)
Comment (trumko) [reply src/сomponents/Experimental/FirstType/FirstType.test.js]: done

File: src/сomponents/Experimental/FirstType/FirstType.test.js (hunk 7 above)
Comment (trumko) [reply src/сomponents/Experimental/FirstType/FirstType.test.js]: Fixed, thank you!

**Review by blyzniuk — APPROVED** [review]

**Pull request merged**